### PR TITLE
GEB-422: HashMap.Values class in Groovy versions before 2.4.0 doesn't have a 'minus' method - converting it to a List to work with Groovy versions before 2.4.0

### PR DESCRIPTION
**Pull request by craigatk**
Background info in https://github.com/geb/issues/issues/422

To fully test that this fix works with Groovy versions < 2.4.0 I had to change the Groovy version in geb-grails.gradle from "project.groovyVersion" (2.4.5) to "2.3.7" (2.3.7 is the Groovy version that ships with Grails 2.4.2 which is Grails version used in geb-grails.gradle). I thought about committing that change but I didn't know if it would affect building/publishing the Grails plugin so I left it out. If you'd prefer that I commit that change just let me know and I'll include it. Or if you know of a way to set the Groovy version for the tests only that might work as well.

### CODE REVIEW

**Comment by craigatk** [comment]
Ahh good point that it's a Groovy compatibility issue and not a Grails one.

Adding a new Groovy compatibility subproject and corresponding test sounds like a fun challenge! I'll give it a try tomorrow but if I run into too many issues I'll probably defer back to you :)

**Comment by erdi** [comment]
Awesome. As I said, I probably won't have time to look into it before Friday evening so you can take your time. Please let me know tomorrow how it went.

File: geb.gradle [hunk 1]
```diff
@@ -153,6 +142,17 @@ subprojects {
         }
     }
 
+    tasks.withType(Test) {
```
Comment (craigatk) [inline geb.gradle]: I needed just the test task configuration section of the groovyModules configuration applied to the new compatibility subproject (the groovyModules section does other things like set an explicit Groovy 2.4.5 dependency that I don't want for the compatibility subproject), so I moved the test task configuration out of the groovyModules block. My Gradle skills aren't the best, so if there is a better way to accomplish this just let me know.

**Comment by craigatk** [comment]
@erdi I've got a first attempt at creating the compatibility subproject with a test to verify Groovy 2.3.7 compatibility. When you get a chance to look at it let me know if there's anything I should do differently. Thanks!

**Comment by erdi** [comment]
This will need some small tweaks but I will take care of that. Thanks, it looks good. I will merge it by the end of the weekend.

**Comment by craigatk** [comment]
Sounds good, thanks Marcin!

**Comment by erdi** [comment]
Thanks again for finding and fixing the issue. I made some changes to the build side of things introduced by this PR. You can see what they were by taking a look at https://github.com/geb/geb/commit/80bb71101a3fb4dc43f9910d399b9b4e8b0fea59 and https://github.com/geb/geb/commit/3f4fd6193586e87f9f81abd92fc2131feda70d77 if you are interested. I will push a minor release containing this fix in the coming days.

**Comment by craigatk** [comment]
Cool I hadn't seen that way to override the versions using the resolution strategy before, that's pretty slick. Thanks for working on a release with this fix!

**Pull request merged**